### PR TITLE
[BUILD] [GA] Support JDK 11 and enable it in GitHub Action

### LINKED ISSUE #550: Enable JDK 11 in GA

### PR DESCRIPTION
**Pull request by pan3793**
### _Why are the changes needed?_

Close #550, relate to #559 #688 #689.

Add `javax.xml.bind:jaxb-api` and `javax.activation:activation` since those apis has been removed by JDK 11.

In GA, we saw a significant performance downgrade in the unit tests, and even with some changes like enlarging timeout, using in-memory catalog instead of hive catalog, the JDK11(~43min) tests cost 2 times than JDK8(~22min), this performance regression is tracked in #745.

### _How was this patch tested?_
- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [ ] [Run test](https://kyuubi.readthedocs.io/en/latest/tools/testing.html#running-tests) locally before make a pull request

### CODE REVIEW

File: .github/workflows/master.yml [hunk 1]
```diff
@@ -32,18 +32,31 @@ jobs:
...
+            spark-hadoop: '2.7'
+            profiles: '-Dspark.archive.mirror=https://archive.apache.org/dist/spark/spark-3.1.2 -Dspark.archive.name=spark-3.1.2-bin-hadoop2.7.tgz -Dmaven.plugin.scalatest.exclude.tags=org.apache.kyuubi.tags.ExtendedSQLTest,org.apache.kyuubi.tags.DeltaTest,org.apache.kyuubi.tags.IcebergTest'
+          - java: 8
+            spark: '3.1'
+            spark-hadoop: '3.2'
+            profiles: '-Pkyuubi-extension-spark_3.1'
+          - java: 11
+            spark: '3.1'
+            spark-hadoop: '3.2'
+            profiles: '-Pkyuubi-extension-spark_3.1 -Djava.version=11'
```
Comment (yaooqinn) [inline .github/workflows/master.yml:50]: How about skip this extension test

**Comment by yaooqinn** [comment]
can you fix the conflicts? and open another issue to track the perf regress for jdk11?

**Comment by pan3793** [comment]
Rebased on master and open a new issue #745 to track the performance regression.

File: kyuubi-common/pom.xml [hunk 2]
```diff
@@ -79,6 +79,11 @@
             <artifactId>hive-service-rpc</artifactId>
         </dependency>
 
+        <dependency>
```
Comment (yaooqinn) [inline kyuubi-common/pom.xml:82]: who/where use this in the common module？

File: kyuubi-main/src/test/scala/org/apache/kyuubi/WithKyuubiServer.scala [hunk 3]
```diff
@@ -46,7 +46,7 @@ trait WithKyuubiServer extends KyuubiFunSuite {
     conf.set(HA_ZK_QUORUM, zkServer.getConnectString)
     conf.set(HA_ZK_ACL_ENABLED, false)
 
-    conf.setIfMissing(ENGINE_INIT_TIMEOUT, 120000L)
+    conf.setIfMissing(ENGINE_INIT_TIMEOUT, 300000L)
```
Comment (yaooqinn) [inline kyuubi-main/src/test/scala/org/apache/kyuubi/WithKyuubiServer.scala:49]: We need add a TODO here to point to the related issue ticket

File: kyuubi-main/src/test/scala/org/apache/kyuubi/operation/KyuubiOperationPerConnectionSuite.scala [hunk 4]
```diff
@@ -42,7 +42,7 @@ class KyuubiOperationPerConnectionSuite extends WithKyuubiServer with JDBCTestUt
       executeStmtReq.setRunAsync(true)
       val executeStmtResp = client.ExecuteStatement(executeStmtReq)
 
-      eventually(timeout(10.seconds), interval(500.milliseconds)) {
+      eventually(timeout(60.seconds), interval(500.milliseconds)) {
```
Comment (yaooqinn) [inline kyuubi-main/src/test/scala/org/apache/kyuubi/operation/KyuubiOperationPerConnectionSuite.scala:45]: ditto

File: kyuubi-main/src/test/scala/org/apache/kyuubi/operation/KyuubiOperationYarnClusterSuite.scala [hunk 5]
```diff
@@ -26,7 +26,7 @@ class KyuubiOperationYarnClusterSuite extends WithKyuubiServerOnYarn with JDBCTe
   override protected def jdbcUrl: String = getJdbcUrl
 
   override protected val kyuubiServerConf: KyuubiConf = {
-    KyuubiConf().set(ENGINE_INIT_TIMEOUT, 300000L)
+    KyuubiConf().set(ENGINE_INIT_TIMEOUT, 600000L)
```
Comment (yaooqinn) [inline kyuubi-main/src/test/scala/org/apache/kyuubi/operation/KyuubiOperationYarnClusterSuite.scala:29]: ditto, it cost 10 minutes to bootstrap

File: pom.xml [hunk 6]
```diff
@@ -63,6 +63,7 @@
         <scala.version>2.12.14</scala.version>
         <scala.binary.version>2.12</scala.binary.version>
 
+        <activation.version>1.1.1</activation.version>
```
Comment (yaooqinn) [inline pom.xml:66]: javax-activation.version

File: pom.xml [hunk 7]
```diff
@@ -1291,6 +1315,7 @@
                         <reportsDirectory>${project.build.directory}/surefire-reports</reportsDirectory>
                         <junitxml>.</junitxml>
                         <filereports>TestSuite.txt</filereports>
+                        <argLine>-Xmx4g -Xss4m -XX:ReservedCodeCacheSize=512M</argLine>
```
Comment (yaooqinn) [inline pom.xml:1318]: -Xmx4g -Xms4g -Xss32m -XX:ReservedCodeCacheSize=512M

File: pom.xml [hunk 8]
```diff
@@ -1570,6 +1595,26 @@
             </modules>
         </profile>
 
+        <profile>
+            <id>java-8</id>
+            <activation>
```
Comment (yaooqinn) [inline pom.xml:1600]: what is this for?

File: kyuubi-common/pom.xml (hunk 2 above)
Comment (pan3793) [reply kyuubi-common/pom.xml:82]: We don't use it. `jaxb-api` should be considered as the transitive deps of `hadoop-client-runtime`. 
If we move `jaxb-api` to test scope, and accident invocation of Hadoop API which direct or indirect requires `jaxb-api`, the ut pass, but will cause runtime failures in production.

File: pom.xml (hunk 8 above)
Comment (pan3793) [reply pom.xml:1600]: auto activate profile by jdk version

File: pom.xml (hunk 7 above)
Comment (pan3793) [reply pom.xml]: After changes, GA failed again, now shrink to `-Xmx2g -Xms2g -Xss8m -XX:ReservedCodeCacheSize=512M` and waiting GA results.

**Review by yaooqinn — APPROVED** [review]

**Comment by pan3793** [comment]
Thanks for reviewing, as all comments are addressed and GA passed, I'm going to merge it for v1.3.0.

**Comment by ulysses-you** [comment]
late lgtm
